src/test/java/com/github/checkstyle/regression/git/DiffParserTest.java

```diff
@@ -203,4 +204,26 @@ public void testParseFilePermissionChange() throws Exception {
             }
         }
     }
+
+    @Test
+    public void testParseLineChanges() throws Exception {
+        try (Repository repository = GitUtils.createNewRepository()) {
+            final File helloWorld = GitUtils.addAnEmptyFileAndCommit(repository, "HelloWorld");
+            Files.write(helloWorld.toPath(), "line 0\nline 1\nline 2\n"
+                    .getBytes(Charset.forName("UTF-8")), StandardOpenOption.APPEND);
+            GitUtils.addAllAndCommit(repository, "add original three lines");
+            GitUtils.createNewBranchAndCheckout(repository, "foo");
+            Files.write(helloWorld.toPath(), "line 0\nline 1 change\nline X\n"
+                    .getBytes(Charset.forName("UTF-8")), StandardOpenOption.WRITE);
+            GitUtils.addAllAndCommit(repository, "change line 1, delete line 2, add a line");
+            final List<GitChange> changes = DiffParser.parse(
+                    repository.getDirectory().getParent(), "foo");
+            assertEquals("There should be 1 change detected", 1, changes.size());
+            final GitChange expected = ImmutableGitChange.builder()
+                    .path("HelloWorld")
+                    .addLines(1, 2)
```

Comment:
I split this UT to three, including adding line, removing line and changing line.